File: Python Scripts/Models/CorporateAction.py
from datetime import datetime
from Models import CorporateActionType

class CorporateAction():
    ticker: str
    timestamp: datetime
    type: CorporateActionType
    value: float
    
    def __init__(self, ticker: str, timestamp: datetime, type: CorporateActionType, value: float) -> None:
        self.ticker = ticker
        self.timestamp = timestamp
        self.type = type
        self.value = value
        
    def to_dict(self):
        return {
            "ticker": self.ticker,
            "timestamp": self.timestamp.isoformat() if isinstance(self.timestamp, datetime) else self.timestamp,
            "type": self.type,
            "value": self.value
        }

File: Core/Models/Models/Common/CorporateAction.cs
using OctoWhirl.Core.Models.Models.Common.Interfaces;
using OctoWhirl.Core.Models.Models.Enums;

namespace OctoWhirl.Core.Models.Models.Common
{
    public abstract class CorporateAction : IMarketData
    {
        public abstract CorporateActionType ActionType { get; }
        public string Instrument { get; set; }
        public DataSource Source { get; set; }
        public DateTime Timestamp { get; set; }
    }
}
